server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -845,24 +848,40 @@ ObjectMapper findParentMapper(String leafFieldPath) {
         return null;
     }
 
-    protected SourceLoader.SyntheticFieldLoader syntheticFieldLoader(Stream<Mapper> mappers, boolean isFragment) {
-        var fields = mappers.sorted(Comparator.comparing(Mapper::fullPath))
-            .map(Mapper::syntheticFieldLoader)
+    SourceLoader.SyntheticFieldLoader syntheticFieldLoader(SourceFilter filter, Collection<Mapper> mappers, boolean isFragment) {
```

Comment:
Nit: keep passing the stream to avoid reconverting the list to a stream? 

I also wonder about the overhead here, shall we skip streams altogether and use good-old collections?